File: legacy/legacy-apis/Psi.Models.ClientConfigurationModels/PersonPaymentSecuritySettings.cs
namespace Psi.Data.Models.ClientConfigurationModels
{
    public class PersonPaymentSecuritySettings : SettingsBaseHelper
    {
        public PersonPaymentSecuritySettings(ISettingsBase settingsBase) : base(settingsBase)
        {
        }

        [SettingKey("PersonPayment.Security.Enabled")]
        public bool Enabled
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("PersonPayment.Security.MinVersion")]
        public double MinVersion
        {
            get => GetDoubleValue();
            set => SetValue(value);
        }

        [SettingKey("PersonPayment.Security.DaysAfterSensitiveInfoChangeAccessRestriction")]
        public double DaysAfterSensitiveInfoChangeAccessRestriction
        {
            get => GetIntValue();
            set => SetValue(value);
        }

        [SettingKey("PersonPayment.Security.SpectrumMemoNumber")]
        public string SpectrumMemoNumber
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("PersonPayment.Security.RestrictAccessFlagNumber")]
        public string RestrictAccessFlagNumber
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("PersonPayment.Security.SymitarRestrictAccessFlagNumber")]
        public int SymitarRestrictAccessFlagNumber
        {
            get => GetIntValue();
            set => SetValue(value);
        }

        [SettingKey("PersonPayment.Security.DnaRestrictAccessFlagNumber")]
        public string DnaRestrictAccessFlagNumber
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("PersonPayment.Security.CoreRestrictAccessFlagNumber")]
        public string CoreRestrictAccessFlagNumber
        {
            get => GetValue();
            set => SetValue(value);
        }

        [SettingKey("PersonPayment.Security.ShouldRestrictAccessAfterAddressChange")]
        public bool ShouldRestrictAccessAfterAddressChange
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("PersonPayment.Security.ShouldRestrictAccessAfterEmailChange")]
        public bool ShouldRestrictAccessAfterEmailChange
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("PersonPayment.Security.ShouldRestrictAccessAfterForgotPassword")]
        public bool ShouldRestrictAccessAfterForgotPassword
        {
            get => GetBoolValue();
            set => SetValue(value);
        }

        [SettingKey("PersonPayment.Security.ShouldRestrictAccessAfterMobileDeviceRegistration")]
        public bool ShouldRestrictAccessAfterMobileDeviceRegistration
        {
            get => GetBoolValue();
            set => SetValue(value);
        }
    }
}

File: workspace/legacy-analyzer/output/infrastructure/models/PersonPaymentSecuritySettings.ts
// Generated imports

export interface PersonPaymentSecuritySettings {
    /** @settingKey PersonPayment.Security.Enabled */
    enabled: boolean;
    /** @settingKey PersonPayment.Security.MinVersion */
    minVersion: number;
    /** @settingKey PersonPayment.Security.DaysAfterSensitiveInfoChangeAccessRestriction */
    daysAfterSensitiveInfoChangeAccessRestriction: number;
    /** @settingKey PersonPayment.Security.SpectrumMemoNumber */
    spectrumMemoNumber: string;
    /** @settingKey PersonPayment.Security.RestrictAccessFlagNumber */
    restrictAccessFlagNumber: string;
    /** @settingKey PersonPayment.Security.SymitarRestrictAccessFlagNumber */
    symitarRestrictAccessFlagNumber: number;
    /** @settingKey PersonPayment.Security.DnaRestrictAccessFlagNumber */
    dnaRestrictAccessFlagNumber: string;
    /** @settingKey PersonPayment.Security.CoreRestrictAccessFlagNumber */
    coreRestrictAccessFlagNumber: string;
    /** @settingKey PersonPayment.Security.ShouldRestrictAccessAfterAddressChange */
    shouldRestrictAccessAfterAddressChange: boolean;
    /** @settingKey PersonPayment.Security.ShouldRestrictAccessAfterEmailChange */
    shouldRestrictAccessAfterEmailChange: boolean;
    /** @settingKey PersonPayment.Security.ShouldRestrictAccessAfterForgotPassword */
    shouldRestrictAccessAfterForgotPassword: boolean;
    /** @settingKey PersonPayment.Security.ShouldRestrictAccessAfterMobileDeviceRegistration */
    shouldRestrictAccessAfterMobileDeviceRegistration: boolean;
}
